lib/patron/session.rb

```diff
@@ -87,12 +87,23 @@ class Session
     private :handle_request, :enable_cookie_session, :set_debug_file
 
     # Create a new Session object.
-    def initialize
-      @headers = {}
-      @timeout = 5
-      @connect_timeout = 1
-      @max_redirects = 5
-      @auth_type = :basic
+    def initialize(args = {}, &block)
+
+      # Allows accessors to be set via constructor hash. Ex:  {:base_url => 'www.home.com'}
+      args.each do |attribute, value|
+        self.send("#{attribute}=", value)
```

Comment:
I know I suggested it, but you might not even need `self` here. I might try it without it.